File: framework/languages/java/samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/play/PlayServerHostFactory.java
package systems.zlink.samples.tictactoe.server.play;

import org.springframework.boot.WebApplicationType;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.Bean;
import systems.zlink.framework.spring.ZLinkFrameworkOptionsCustomizer;
import systems.zlink.samples.tictactoe.server.configuration.SampleSettings;

@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackageClasses = PlayServer.class)
public final class PlayServerHostFactory {
    private PlayServerHostFactory() {
    }

    public static ConfigurableApplicationContext start(SampleSettings settings) {
        SpringApplicationBuilder builder = new SpringApplicationBuilder(PlayServerHostFactory.class)
            .web(WebApplicationType.NONE)
            .initializers(context ->
                context.getBeanFactory().registerSingleton("sampleSettings", settings));
        builder.application().setKeepAlive(true);
        return builder.run();
    }

    @Bean
    ZLinkFrameworkOptionsCustomizer playOptions(SampleSettings settings) {
        return PlayServer.configure(settings);
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/play/PlayServerHostFactory.kt
package systems.zlink.samples.kotlin.tictactoe.server.play

import org.springframework.boot.WebApplicationType
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.context.annotation.Bean
import systems.zlink.framework.spring.ZLinkFrameworkOptionsCustomizer
import systems.zlink.samples.kotlin.tictactoe.server.configuration.SampleSettings
import systems.zlink.samples.kotlin.tictactoe.server.play.gamespots.handlers.TicTacToeGameCreatedHandler

@SpringBootApplication(
    proxyBeanMethods = false,
    scanBasePackageClasses = [PlayServer::class],
)
class PlayServerHostFactory {
    @Bean
    fun playOptions(settings: SampleSettings): ZLinkFrameworkOptionsCustomizer =
        PlayServer.configure(settings)

    @Bean
    fun ticTacToeGameCreatedHandler(): TicTacToeGameCreatedHandler =
        TicTacToeGameCreatedHandler()

    companion object {
        fun start(settings: SampleSettings): ConfigurableApplicationContext =
            SpringApplicationBuilder(PlayServerHostFactory::class.java).also { builder ->
                builder.application().setKeepAlive(true)
            }
                .web(WebApplicationType.NONE)
                .initializers(ApplicationContextInitializer<ConfigurableApplicationContext> { context ->
                    context.beanFactory.registerSingleton("sampleSettings", settings)
                })
                .run()
    }
}
